docs/pipeline.md

```diff
@@ -32,115 +33,171 @@ def build_pipeline():
-when KFP is setup on GCP. 
+## Compiling and Running a pipeline
 
-More info here: https://v1-6-branch.kubeflow.org/docs/distributions/gke/pipelines/preemptible/
+Once all your components are added to your pipeline you can use different compilers to run your pipeline:
 
+!!! note "IMPORTANT"
+  When using other runners you will need to make sure that your new environment has access to:
+  - The base path of your pipeline (as mentioned above)
+  - The images used in your pipeline (make sure you have access to the registries where the images are stored)
 
+### Local Runner
 
-## Setting Custom partitioning parameters
+This local runner is mainly aimed at local development and quick iterations, there is no scaling so using small slices of your data is advised.
 
-When working with Fondant, each component deals with datasets, and Dask is used internally 
-to handle datasets larger than the available memory. To achieve this, the data is divided 
-into smaller chunks called "partitions" that can be processed in parallel. Ensuring a sufficient number of partitions
-enables parallel processing, where multiple workers process different partitions simultaneously, 
-and smaller partitions ensure they fit into memory.
+The local compiler will take your pipeline and create a `docker-compose.yml` file where every component is added as a service with the correct dependencies by leveraging the `depends_on` functionality and the `service_completed_successfully` status. See the basic example below:
 
-### How Fondant handles partitions
+```yaml
+version: '3.8'
+services:
+  component_1:
+    command: ["python", "main.py"]
+    image: component_1:latest
+  component_2:
+    command: ["python", "main.py"]
+    image: component_2:latest
+    depends_on:
+      component_1:
+        condition: service_completed_successfully
+  component_3:
+    command: ["python", "main.py"]
+    depends_on:
+      component_2:
+        condition: service_completed_successfully
+    build: ./component_3
+```
 
-**1) Repartitioning the Loaded DataFrame:** This step is optional and comes into play if the number
-of partitions is fewer than the available workers on the data processing instance.
-By repartitioning, the maximum number of workers can be efficiently utilized, leading to faster
-and parallel processing.
+Note that for components that do not come from the registry (local custom components) the compiler will add a build subsection (see component_3 in the example above) instead of referring to the image specified in the `component_spec.yaml`. This allows docker-compose to build and rebuild the container used in that component allowing for quicker iteration.
 
-**2) Repartitioning the Written DataFrame:** The written dataframe is also repartitioned into
-smaller sizes (default 250MB) to enable the next component to load these partitions into memory.
+In order to compile your pipeline to a `docker-compose` spec you need to import the `DockerCompiler`
```

Comment:
Should we document the Python API here? I would focus on using the CLI, and including it here might make it unclear if they need to add this to their code somewhere.

Reply:
I would even just document the `run` command, not the `compile` command.

Reply:
Same for the other runners below.